apps/dashboard/lib/smart_attributes/attributes/auto_modules.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module SmartAttributes
+  class AttributeFactory
+    # Build this attribute object with defined options
+    # @param opts [Hash] attribute's options
+    # @return [Attributes::AutoModules] the attribute object
+    def self.build_auto_modules(opts = {})
+      Attributes::AutoModules.new('auto_modules', opts)
+    end
+  end
+
+  module Attributes
+    # AutoModules populates a select widget of modules from HpcModule class
+    # that is cluster aware. Meaning it will attach data-option-for-cluster-X
+    # attributes to the options.
+    class AutoModules < Attribute
+      def initialize(id, opts = {})
```

Comment:
I'm not sure what the `id` being initialized here is being used for.

Reply:
Looking above, it looks like `auto_modules` is passed as the `id`, but I don't see how this then gets used within this AutoModules class.

Reply:
`id` is like, well the id ends up being the html id. I think other classes use it. The factory passes it in, look at the parent class.

Reply:
There's a bug here actually where since the `id` wasn't unique it wouldn't dynamically update if there were 2, good catch!